drivers/soundwire/intel.c

```diff
@@ -1544,15 +1545,20 @@ static int intel_master_remove(struct platform_device *pdev)
 	return 0;
 }
 
-int intel_master_process_wakeen_event(struct platform_device *pdev)
+static void intel_link_shutdown(struct ancillary_device *ancildev)
 {
-	struct device *dev = &pdev->dev;
+	intel_link_remove(ancildev);
```

Comment:
Do we need to handle the case when intel_link_shutdown() is invoked before intel_link_remove()? In that case intel_link_remove() will be triggered twice without intel_link_probe(). I can see the case when I run the "sudo poweroff" command. And I will see "error: hda_dsp_core_power_down: timeout on HDA_DSP_REG_ADSPCS read" in the case.

Reply:
@bardliao I must admit I don't understand what shutdown means, and if we have a problem here we had it before with platform devices.

@ranj063 any suggestions on how we might deal with this?

Reply:
I dont quite understand what it means by "intel_link_shutdown() is invoked before intel_link_remove()?" Why would remove() be invoked after shutdown?

Reply:
Sorry, I made a silly mistake here. intel_link_remove() is called by intel_link_shutdown() here. The "error: hda_dsp_core_power_down: timeout on HDA_DSP_REG_ADSPCS read" is only seen in the SDCA device. It must be a different issue.